src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -1576,6 +1638,32 @@ public virtual ConcurrentDictionary<string, PackageResult> Upgrade(ChocolateyCon
                             {
                                 if (packageToUninstall != null)
                                 {
+                                    // Are we attempting a downgrade? We need to ensure it's allowed...
+                                    if (!config.AllowDowngrade && packageToUninstall.Identity.HasVersion && packageDependencyInfo.HasVersion && packageDependencyInfo.Version < packageToUninstall.Identity.Version)
+                                    {
+                                        var logMessage = StringResources.ErrorMessages.UnableToDowngrade.FormatWith(packageToUninstall.Name, packageToUninstall.Version, Environment.NewLine);
+                                        var nullResult = packageResultsToReturn.GetOrAdd(packageToUninstall.Name, packageToUninstall);

```

Comment:
Once again here.

Reply:
Same as before, removed `nullResult` where not needed.